app.rb

```diff
@@ -0,0 +1,142 @@
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def intro
+    puts 'Welcome to School Library App!'
+  end
+
+  def display_options
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  # option 1 - List all books
+  def list_books
+    puts 'No book yet!' if @books.empty?
+    @books.each { |book| puts "Title: \"#{book.title}\", Author: #{book.author}" }
+  end
+
+  # option 2 - List all people
+  def list_people
+    puts 'We do not have people yet' if @people.empty?
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  # Create a student
+  def create_student
+    print 'Age: '
+    age = gets.to_i
+    print "\nName: "
+    name = gets.chomp.capitalize
+    print "\nHas parent permission? [Y/N]: "
+    permission = gets.downcase == 'y'
+    @people << Student.new(age, name, permission)
+    puts 'Person created successfully'
+  end
+
+  # Create a teacher
+  def create_teacher
+    print 'Age: '
+    age = gets.to_i
+    print "\nName: "
+    name = gets.chomp.capitalize
+    print "\nSpecialization: "
+    specialization = gets.chomp.capitalize
+    @people << Teacher.new(specialization, age, name)
+    puts 'Person created successfully'
+  end
+
+  # option 3 - Create a person
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    option = gets.chomp
+    if option == '1'
+      create_student
+    elsif option == '2'
+      create_teacher
+    else
+      puts "You entered a wrong input, it's either 1 (for a new student) or 2 (for a new teacher)"
+    end
+  end
+
+  # option 4 - Create a book
+  def create_book
+    print "\nTitle: "
+    title = gets.chomp.capitalize
+    print "\nAuthor: "
+    author = gets.chomp.capitalize
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  # option 5 - Create a rental
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'There are no books or people to create a rental'
+      return
+    end
+    puts 'Select a book from the following list by number (not by id)'
+    @books.each_with_index { |book, index| puts "#{index}) Title: \"#{book.title}\", Author: \"#{book.author}\"" }
+    book_id = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not by id)'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_id = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_id], @people[person_id])
+    puts 'Rental created successfully'
+  end
```

Comment:
![image](https://github.com/kkmanuu/OOP-school-library/assets/93039359/04602667-9f27-4815-a338-3fc0c92167ac)

- [ ] It would be nice if you see this issue :eyes: when I try to create a rental then I don't see any books but I have books in my database but they are not showing up while creating the rentals :fearful: Kindly fix this issue as well to make your app work without any errors :+1:

Reply:
@HAMMAS-SALEEM Done making the changes